### PR TITLE
Add more validators to BitcrowdEcto.Changeset

### PR DESCRIPTION
**Pull request by agatheblues**
There's a few more validators that we could add to the project:

- validate_order
- validate_datetime_order
- validate_date_order
- validate_datetime_after
- validate_future_datetime

They vary a little from the original ones, as they get a `formatter` parameter: that way one can configure how to convert date times to readable strings 🕐

### CODE REVIEW

File: lib/bitcrowd_ecto/changeset.ex [hunk 1]
```diff
@@ -112,4 +112,137 @@ defmodule BitcrowdEcto.Changeset do
...
+  @doc """
+  Validates a field timestamp to be after the given one
+  """
+  @doc since: "0.6.0"
+  @spec validate_datetime_after(Ecto.Changeset.t(), atom, DateTime.t(), fun) :: Ecto.Changeset.t()
+  def validate_datetime_after(
+        changeset,
+        field,
+        reference_datetime,
+        formatter \\ &DateTime.to_string/1
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:155]: i feel like this should be a keyword option, wdyt?

File: lib/bitcrowd_ecto/changeset.ex [hunk 2]
```diff
@@ -112,4 +112,137 @@ defmodule BitcrowdEcto.Changeset do
...
+  """
+  @doc since: "0.6.0"
+  @spec validate_datetime_order(Ecto.Changeset.t(), atom, atom, list(atom), fun) ::
+          Ecto.Changeset.t()
+  def validate_datetime_order(
+        changeset,
+        from_field,
+        until_field,
+        valid_orders \\ [:lt, :eq],
+        formatter \\ &to_local_datetime/2
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:200]: 🤔  hmm this `to_local_datetime` thing is a bit odd, IMO. If you absolutely want to include it, maybe move it to `BitcrowdEcto.DateTime`... But perhaps just default to `DateTime.to_string`? It seems oddly specific with the timezone shift, etc.

File: lib/bitcrowd_ecto/changeset.ex [hunk 3]
```diff
@@ -112,4 +112,137 @@ defmodule BitcrowdEcto.Changeset do
...
+      until_field,
+      :datetime_order,
+      &DateTime.compare/2,
+      valid_orders,
+      formatter
+    )
+  end
+
+  @doc since: "0.6.0"
+  @spec validate_order(Ecto.Changeset.t(), atom, atom, atom, fun, list(atom), fun) ::
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:214]: this is missing a `@doc` string

File: lib/bitcrowd_ecto/changeset.ex [hunk 4]
```diff
@@ -112,4 +112,137 @@ defmodule BitcrowdEcto.Changeset do
...
+          Ecto.Changeset.t()
+  def validate_order(
+        changeset,
+        from_field,
+        until_field,
+        validation_key,
+        compare_fun \\ &Kernel.</2,
+        valid_orders \\ [true],
+        formatter \\ &to_local_datetime/2
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:223]: 🤔  if `validate_order` is a generic "make sure a is less than b" validator, `to_local_datetime` shouldn't be the default, but `Kernel.to_string/1`, IMO. Also, the `case` block below for the `stringified` value should go away, doesn't make sense for this, I think.

**Review by maltoe — CHANGES_REQUESTED** [review]
👍  Thank you for doing this @agatheblues 💜  Marked a few things that I hink we should improve before allowing this to become the API of the library.

File: lib/bitcrowd_ecto/changeset.ex (hunk 1 above)
Comment (agatheblues) [reply lib/bitcrowd_ecto/changeset.ex]: Right! I also moved `valid_orders` and `compare_fun` to be part of the `opts`.

File: lib/bitcrowd_ecto/changeset.ex (hunk 2 above)
Comment (agatheblues) [reply lib/bitcrowd_ecto/changeset.ex]: Indeed, and that's solved by turning the formatter into a simpler form in `validate_order` 👍

File: lib/bitcrowd_ecto/changeset.ex [hunk 5]
```diff
@@ -112,4 +112,134 @@ defmodule BitcrowdEcto.Changeset do
...
+          Ecto.Changeset.t()
+  def validate_datetime_after(changeset, field, reference_datetime, opts \\ []) do
+    formatter = Keyword.get(opts, :formatter, &DateTime.to_string/1)
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(reference_datetime, datetime) != :lt do
+      add_error(
+        changeset,
+        field,
+        "must be after #{formatter.(reference_datetime)}",
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:160]: Sorry for nitpicking again, but I didn't spot this in the previous review: I think this is going through Gettext in the web layer, right? In that case, it should be

```suggestion
        "must be after %{reference}",
        reference: formatter.(reference_datetime),
```

File: lib/bitcrowd_ecto/changeset.ex [hunk 6]
```diff
@@ -112,4 +112,134 @@ defmodule BitcrowdEcto.Changeset do
...
+      compare_fun: &DateTime.compare/2,
+      valid_orders: valid_orders,
+      formatter: formatter
+    )
+  end
+
+  @doc """
+  Validates two fields to be a range, so if both are set the first has to be before
+  the second field. The error is placed on the second field.
+  """
```
Comment (maltoe) [inline lib/bitcrowd_ecto/changeset.ex:221]: some examples maybe? 🙏

File: lib/bitcrowd_ecto/changeset.ex (hunk 5 above)
Comment (agatheblues) [reply lib/bitcrowd_ecto/changeset.ex:160]: 👍 In that case I have to drop my tests on passing a custom formatter! Or is there a way to get the error message as interpreted by gettext?

Comment (agatheblues) [reply lib/bitcrowd_ecto/changeset.ex:160]: Nevermind what I just said 🔪

**Review by maltoe — APPROVED** [review]
LGTM 🚢

**Pull request merged**